GliaWidgets/Sources/View/Chat/GVA/QuickReply/QuickReplyView.swift

```diff
@@ -0,0 +1,114 @@
+import UIKit
+
+final class QuickReplyView: UIView {
+
+    var props: Props = .hidden {
+        didSet {
+            renderProps()
+        }
+    }
+
+    let style: GvaQuickReplyButtonStyle
+    private lazy var collectionView: SelfSizingCollectionView = {
+        let layout = LeftAlignedCollectionViewFlowLayout()
+        layout.estimatedItemSize = UICollectionViewFlowLayout.automaticSize
+        layout.minimumLineSpacing = 0
+        layout.minimumInteritemSpacing = 12
+        layout.sectionInset = .init(top: 8, left: 16, bottom: 8, right: 16)
+        return SelfSizingCollectionView(
+            frame: .zero,
+            collectionViewLayout: layout
+        )
+    }()
+    private var collectionViewHeightConstraint: NSLayoutConstraint?
+
+    init(style: GvaQuickReplyButtonStyle) {
+        self.style = style
+        super.init(frame: .zero)
+        setUp()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - Private
+
+private extension QuickReplyView {
+    func renderProps() {
+        switch props {
+        case .shown:
+            collectionViewHeightConstraint?.priority = .defaultLow
+        case .hidden:
+            collectionViewHeightConstraint?.priority = .required
+        }
+        collectionView.reloadData()
+    }
+
+    func setUp() {
+        collectionView.register(
+            QuickReplyButtonCell.self,
+            forCellWithReuseIdentifier: QuickReplyButtonCell.identifier
+        )
+        collectionView.dataSource = self
+
+        addSubview(collectionView)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+
+        let height = collectionView.heightAnchor.constraint(equalToConstant: 0)
+        height.priority = .required
+        collectionViewHeightConstraint = height
+
+        NSLayoutConstraint.activate([
```

Comment:
if you will inherit from BaseView, you can establish constraints in defineLayout function.

Reply:
Fixed 0270b3ef7b17e734cbe3424c77ed8fe16513f45f